packages/odyssey-storybook/src/components/odyssey-labs/DatePicker/DatePicker.stories.tsx

```diff
@@ -12,11 +12,79 @@
 
 import { useMemo, useState } from "react";
 import type { Meta, StoryObj } from "@storybook/react";
+import { screen, userEvent, waitFor, within } from "@storybook/testing-library";
+import { expect } from "@storybook/jest";
 
 import { DatePicker, DatePickerProps } from "@okta/odyssey-react-mui/labs";
 
 import { fieldComponentPropsMetaData } from "../../../fieldComponentPropsMetaData";
 import { MuiThemeDecorator } from "../../../../.storybook/components";
+import { axeRun } from "../../../axe-util";
+import type { PlaywrightProps } from "../../odyssey-mui/storybookTypes";
+
+const findInputElement = async ({
+  canvasElement,
+  step,
+}: PlaywrightProps<DatePickerProps>) => {
+  await step("Find Input", async () => {
+    const canvas = within(canvasElement);
+    const input = canvas.getByPlaceholderText("MM/DD/YYYY") as HTMLInputElement;
+    await userEvent.click(input);
+  });
+};
+
+const clickCalendarButton = async ({
+  canvasElement,
+  step,
+}: PlaywrightProps<DatePickerProps>) => {
+  await step("Click Calendar Button", async () => {
+    const canvas = within(canvasElement);
+    const buttonElement = canvas.getByLabelText("Choose date");
+    await userEvent.click(buttonElement);
+  });
+};
+
+const openCalendarAndSelectDate =
+  ({ canvasElement, step }: PlaywrightProps<DatePickerProps>) =>
+  async (actionName: string) => {
+    await step("select date", async ({ args }) => {
+      // I don't know why this doesn't work without this step
+      await findInputElement({ canvasElement, step });
+      await clickCalendarButton({ canvasElement, step });
+      await waitFor(() => {
+        axeRun(actionName);
+      });
```

Comment:
Why is `axeRun` in a `waitFor`? The `waitFor` function sets a timeout and re-runs the function you give it at an interval. It will keep running it until it stops failing, then it will give you the response.

If it fails by the time the timeout ends, it throws an error.